homeassistant/config_entries.py

```diff
@@ -727,12 +730,28 @@ def async_start_reauth(
         data: dict[str, Any] | None = None,
     ) -> None:
         """Start a reauth flow."""
+        # We will check this again in the task when we hold the lock,
+        # but we also check it now to try to avoid creating the task.
         if any(self.async_get_active_flows(hass, {SOURCE_REAUTH})):
             # Reauth flow already in progress for this entry
             return
```

Comment:
We could just skip this check to simplify the code, or are reauth flows started very often?

Reply:
for the esphome case above they get started frequently since we keep trying to connect over and over to see if the key has changed